src/main/java/com/aws/greengrass/certificatemanager/certificate/CertificateGenerator.java

```diff
@@ -59,9 +60,37 @@ public abstract void generateCertificate(Supplier<List<String>> connectivityInfo
      * @return true if certificate should be regenerated, else false
      */
     public boolean shouldRegenerate() {
-        Instant dayFromNow = Instant.now(clock).plus(1, ChronoUnit.DAYS);
-        Instant expiryTime = getExpiryTime();
-        return expiryTime.isBefore(dayFromNow);
+        return isExpired() || isAboutToExpire();
+    }
+
+    /**
+     * Check if the certificate is expired.
+     *
+     * @return true if the certificate is expired
+     */
+    public boolean isExpired() {
+        return getExpiryTime().isBefore(Instant.now(clock));
+    }
+
+    /**
+     * Check if the certificate is close to expiring.
+     *
+     * @return true if the certificate is to its expiration
+     */
+    public boolean isAboutToExpire() {
```

Comment:
This doesn't seem like the right place for this. Perhaps a slight refactor is a decent idea. The single responsibility of `CertificateGenerator` should be to generate certificates. Anything relating to expiration and making re-generation decisions should be in the expiry monitor. Thoughts?

Reply:
Yeah the expiry monitor feels like a better home, will refactor 👍

Reply:
done, also refactored expiry monitor tests